src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,101 @@
+package duke.task;
+
+import duke.exceptions.UnclearInstructionException;
+import duke.task.Task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline class which encapsulates task's deadline.
+ */
+public class Deadline extends Task {
+    private String byString;
+    private LocalDate byDate;
+
+    /**
+     * Constructor method of Deadline.
+     *
+     * @param description Description of a deadline.
+     * @param by Due time of a deadline.                   
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.byDate = LocalDate.parse(by);
+    }
+
+    /**
+     * Another constructor method of Deadline.
+     *
+     * @param isDone Done indicator of a deadline.  
+     * @param description Description of a deadline.
+     * @param by Due time of a deadline.                   
+     */
+    public Deadline(String isDone, String description, String by) {
```

Comment:
You are using `isDone` string as a boolean type. Perhaps consider using a boolean instead?

Reply:
Agreed.